src/pal/src/file/path.cpp

```diff
@@ -1140,7 +1159,7 @@ SearchPathA(
         if (dw == 0 || dw >= MAX_LONGPATH) 
```

Comment:
In fact, there are more checks to the MAX_LONGPATH in the code that you have not modified and we should take care of all of them.